File: examples/python/apigateway_create_api.py
"""Crea una API REST vacía en API Gateway.

Uso: python3 apigateway_create_api.py [nombre-api]
"""
import sys

import boto3

apigateway = boto3.client(
    "apigateway",
    endpoint_url="http://localhost:4566",
    aws_access_key_id="test",
    aws_secret_access_key="test",
    region_name="us-east-1",
)


def main():
    api_name = sys.argv[1] if len(sys.argv) > 1 else "mi-api"

    result = apigateway.create_rest_api(name=api_name)
    print("API REST creada:", api_name)
    print("API ID:", result["id"])
    print("Siguiente paso: python3 apigateway_create_resource.py", result["id"])


if __name__ == "__main__":
    main()

File: examples/rust/apigateway_create_api.rs
// Crea una API REST vacia en API Gateway.
// Requiere crates: aws-config, aws-sdk-apigateway, aws-credential-types, tokio.
// Uso: cargo run --bin apigateway_create_api -- [nombre-api]

use aws_config::BehaviorVersion;
use aws_credential_types::Credentials;
use aws_sdk_apigateway::{config::Region, Client, Config};

#[tokio::main]
async fn main() -> Result<(), aws_sdk_apigateway::Error> {
    let args: Vec<String> = std::env::args().collect();
    let api_name = args.get(1).cloned().unwrap_or_else(|| "mi-api".to_string());

    let config = Config::builder()
        .behavior_version(BehaviorVersion::latest())
        .endpoint_url("http://localhost:4566")
        .region(Region::new("us-east-1"))
        .credentials_provider(Credentials::new("test", "test", None, None, "floci"))
        .build();
    let client = Client::from_conf(config);

    let result = client.create_rest_api().name(&api_name).send().await?;
    let api_id = result.id().unwrap_or("?");
    println!("API REST creada: {api_name}");
    println!("API ID: {api_id}");
    println!("Siguiente paso: cargo run --bin apigateway_create_resource -- {api_id}");
    Ok(())
}
